hikari/internal/deprecation.py

```diff
@@ -99,3 +99,16 @@ def warn_deprecated(
         category=DeprecationWarning,
         stacklevel=stack_level,
     )
+
+
+if typing.TYPE_CHECKING:
+    from typing_extensions import deprecated
+
+else:
+
+    def deprecated(*args):
+        """Mark a function, overload, or class as deprecated for type-checkers.
+
+        This has no runtime side-effects.
+        """
+        return lambda value: value
```

Comment:
```suggestion
    def deprecated(func):
        """Mark a function, overload, or class as deprecated for type-checkers.

        This has no runtime side-effects.
        """
        return func
```
And make use of ParamSpec?

Reply:
This also decorates classes and paramspec won't work for that and isn't necessary (the return type is just `Callable[[_T], _T]` + this isn't being type checked since its essentially in a `if not typing.TYPE_CHECKING` block